analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -170,12 +177,12 @@ def perform_analysis(args, skip_handler, context, actions, metadata,
     available_checkers = set()
     # Add profile names to the checkers list so we will not warn
     # if a profile is enabled but there is no checker with that name.
-    available_checkers.update(context.available_profiles.keys())
+    available_checkers.update(list(context.available_profiles.keys()))
 
     # Collect all the available checkers from the enabled analyzers.
-    for analyzer in config_map.items():
+    for analyzer in list(config_map.items()):
```

Comment:
Why do we need to eagerly create a list here? (and other places?)

Reply:
yes, you are right the list conversion is not needed here
2to3 made some extra conversion which is probably not needed, unfortunately there are some places where it is needed because the surrounding source code depends on a list, that could be optimized later

I will recheck where can we remove the list conversion safely